mkosi/archive.py

```diff
@@ -102,3 +104,21 @@ def make_cpio(src: Path, dst: Path, files: Optional[Iterable[Path]] = None) -> N
         # Make sure tar uses user/group information from the root directory instead of the host.
         options=finalize_passwd_mounts(dst),
     )
+
+
+def normalize_mtime(root: Path, source_date_epoch: Optional[str]) -> None:
+    if source_date_epoch is None:
+        return
+
+    try:
+        mtime_epoch = int(source_date_epoch)
+    except ValueError:
+        die(f"Invalid SOURCE_DATE_EPOCH value: {source_date_epoch}")
+    if mtime_epoch < 0:
+        die(f"Invalid SOURCE_DATE_EPOCH value: {source_date_epoch}")
+
+    mtime_epoch_ns = mtime_epoch * 10**9
+
+    with complete_step("Normalizing mtime"):
+        for p in root.rglob("*"):
+            os.utime(p, (mtime_epoch_ns, mtime_epoch_ns), follow_symlinks=False)
```

Comment:
The signature of `os.utime` is
```python
def os.utime(path, times=None, *, [ns, ]dir_fd=None, follow_symlinks=True)
```
so `(mtime_epoch_ns, mtime_epoch_ns)` here is `time` instead of `ns` and the docs say
> If times is not None, it must be a 2-tuple of the form (atime, mtime) where each member is an int or float expressing seconds.

so what you probably want is
```suggestion
            os.utime(p, ns=(mtime_epoch_ns, mtime_epoch_ns), follow_symlinks=False)
```

Reply:
I guess we misunderstood the function signature there. I've updated to only use seconds.